aten/src/ATen/native/cpu/LinearAlgebraKernel.cpp

```diff
@@ -79,8 +81,50 @@ void addr_kernel(TensorIterator &iter,
   );
 }
 
+template <typename scalar_t, typename acc_t=typename scalar_value_type<scalar_t>::type>
+void linalg_vector_norm_kernel_cpu_impl(TensorIterator& iter, Scalar val) {
+  double ord_val;
+  if (val.isIntegral(/*includeBool=*/false)) {
+     ord_val = val.to<int64_t>();
+  } else if (val.isFloatingPoint()) {
+     ord_val = val.to<double>();
+  } else {
+     AT_ERROR("linalg_vector_norm_kernel_cuda_impl expects norm to be integer or float");
```

Comment:
`AT_ERROR -> TORCH_CHECK(false,`

Reply:
However, should this be an internal assertion or can it be caused by a user? If it can be caused by the user, then the error message should be updated to not refer to an internal function name and the macro should be TORCH_CHECK, not AT_ERROR or TORCH_INTERNAL_ASSERT.

Reply:
Similar comment as below. Also see comment about use of the word "norm."

Reply:
There isn't any way for this error to be triggered by the user--only by a contributor. The error is only triggered if the `linalg_vector_norm_stub` call in aten/src/ATen/native/LinearAlgebra.cpp is given neither an integer nor float type `ord`. At the moment, `ord` is always converted to a double for this call.